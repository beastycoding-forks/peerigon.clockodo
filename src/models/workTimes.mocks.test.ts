import {
  createChangeRequestMocks,
  createWorkTimeDayMocks,
} from "./workTimes.mocks.js";

describe("workTimesMocks", () => {
  test("createWorkDayMocks", () => {
    expect(
      createWorkTimeDayMocks({
        count: 5,
      })
    ).toMatchSnapshot();
  });

  test("createChangeRequestMocks", () => {
    expect(
      createChangeRequestMocks({
        count: 5,
      })
    ).toMatchSnapshot();
  });
});
