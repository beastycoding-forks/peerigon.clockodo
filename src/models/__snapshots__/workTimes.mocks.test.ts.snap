// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createWorkTimesMocks Beantragte 1`] = `
Array [
  Object {
    "clocking": false,
    "date": "2020-03-24",
    "duration": 17903,
    "id": 0,
    "intervals": Array [
      Object {
        "timeSince": "2020-03-24T15:24:20.000Z",
        "timeUntil": "2020-03-24T16:25:46.000Z",
      },
      Object {
        "timeSince": "2020-03-24T17:28:08.000Z",
        "timeUntil": "2020-03-24T17:43:43.000Z",
      },
      Object {
        "timeSince": "2020-03-24T18:43:19.000Z",
        "timeUntil": "2020-03-24T22:24:41.000Z",
      },
    ],
    "status": 1,
    "usersId": 0,
  },
  Object {
    "clocking": false,
    "date": "2020-04-14",
    "duration": 12772,
    "id": 1,
    "intervals": Array [
      Object {
        "timeSince": "2020-04-14T12:05:31.000Z",
        "timeUntil": "2020-04-14T13:03:41.000Z",
      },
      Object {
        "timeSince": "2020-04-14T15:35:44.000Z",
        "timeUntil": "2020-04-14T18:10:26.000Z",
      },
    ],
    "status": 1,
    "usersId": 0,
  },
  Object {
    "clocking": false,
    "date": "2020-06-05",
    "duration": 13547,
    "id": 2,
    "intervals": Array [
      Object {
        "timeSince": "2020-06-05T13:50:29.000Z",
        "timeUntil": "2020-06-05T14:13:14.000Z",
      },
      Object {
        "timeSince": "2020-06-05T14:39:21.000Z",
        "timeUntil": "2020-06-05T14:43:25.000Z",
      },
      Object {
        "timeSince": "2020-06-05T16:25:38.000Z",
        "timeUntil": "2020-06-05T19:44:36.000Z",
      },
    ],
    "status": 1,
    "usersId": 0,
  },
  Object {
    "clocking": false,
    "date": "2020-09-11",
    "duration": 28176,
    "id": 3,
    "intervals": Array [
      Object {
        "timeSince": "2020-09-11T10:25:10.000Z",
        "timeUntil": "2020-09-11T11:46:26.000Z",
      },
      Object {
        "timeSince": "2020-09-11T16:09:25.000Z",
        "timeUntil": "2020-09-11T22:37:45.000Z",
      },
    ],
    "status": 1,
    "usersId": 0,
  },
  Object {
    "clocking": false,
    "date": "2020-09-17",
    "duration": 18016,
    "id": 4,
    "intervals": Array [
      Object {
        "timeSince": "2020-09-17T14:53:25.000Z",
        "timeUntil": "2020-09-17T19:53:41.000Z",
      },
    ],
    "status": 1,
    "usersId": 0,
  },
]
`;

exports[`createWorkTimesMocks Genehmigte 1`] = `
Array [
  Object {
    "clocking": false,
    "date": "2020-03-24",
    "duration": 17903,
    "id": 0,
    "intervals": Array [
      Object {
        "timeSince": "2020-03-24T15:24:20.000Z",
        "timeUntil": "2020-03-24T16:25:46.000Z",
      },
      Object {
        "timeSince": "2020-03-24T17:28:08.000Z",
        "timeUntil": "2020-03-24T17:43:43.000Z",
      },
      Object {
        "timeSince": "2020-03-24T18:43:19.000Z",
        "timeUntil": "2020-03-24T22:24:41.000Z",
      },
    ],
    "status": 2,
    "usersId": 0,
  },
  Object {
    "clocking": false,
    "date": "2020-04-14",
    "duration": 12772,
    "id": 1,
    "intervals": Array [
      Object {
        "timeSince": "2020-04-14T12:05:31.000Z",
        "timeUntil": "2020-04-14T13:03:41.000Z",
      },
      Object {
        "timeSince": "2020-04-14T15:35:44.000Z",
        "timeUntil": "2020-04-14T18:10:26.000Z",
      },
    ],
    "status": 2,
    "usersId": 0,
  },
  Object {
    "clocking": false,
    "date": "2020-06-05",
    "duration": 13547,
    "id": 2,
    "intervals": Array [
      Object {
        "timeSince": "2020-06-05T13:50:29.000Z",
        "timeUntil": "2020-06-05T14:13:14.000Z",
      },
      Object {
        "timeSince": "2020-06-05T14:39:21.000Z",
        "timeUntil": "2020-06-05T14:43:25.000Z",
      },
      Object {
        "timeSince": "2020-06-05T16:25:38.000Z",
        "timeUntil": "2020-06-05T19:44:36.000Z",
      },
    ],
    "status": 2,
    "usersId": 0,
  },
  Object {
    "clocking": false,
    "date": "2020-09-11",
    "duration": 28176,
    "id": 3,
    "intervals": Array [
      Object {
        "timeSince": "2020-09-11T10:25:10.000Z",
        "timeUntil": "2020-09-11T11:46:26.000Z",
      },
      Object {
        "timeSince": "2020-09-11T16:09:25.000Z",
        "timeUntil": "2020-09-11T22:37:45.000Z",
      },
    ],
    "status": 2,
    "usersId": 0,
  },
  Object {
    "clocking": false,
    "date": "2020-09-17",
    "duration": 18016,
    "id": 4,
    "intervals": Array [
      Object {
        "timeSince": "2020-09-17T14:53:25.000Z",
        "timeUntil": "2020-09-17T19:53:41.000Z",
      },
    ],
    "status": 2,
    "usersId": 0,
  },
]
`;
